File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    static int count = 0;
    public int solution(int[] numbers, int target) {
       
        makeNumbers(numbers, target, 0,0);
        
        return count;
    }
    
    private static void makeNumbers(
        int[] numbers, int target, int idx, int result){
        if(idx == numbers.length){
            if(result == target){
                count++;
            }
            return;
        }
        
        makeNumbers(numbers, target,  idx + 1, result + numbers[idx]);
        makeNumbers(numbers, target,  idx + 1, result - numbers[idx]);
        
    }
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    var answer = 0;
    
    function makeTarget(numbers, target, idx, result){
        if(idx === numbers.length){
            if(result === target) answer++;
            return;
        }
        
        makeTarget(numbers, target, idx+1, result+numbers[idx]);
        makeTarget(numbers, target, idx+1, result-numbers[idx]);
    }   
    
    makeTarget(numbers, target, 0, 0);

    
    return answer;
}


